mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/state/Dsl.kt

```diff
@@ -67,89 +68,247 @@ data class Facades(
     val settings: SettingsFacade,
 )
 
+/** An interface representing a standard testing environment. */
+interface TestEnvironment {
+
+  /** The [Facades] used. */
+  val facades: Facades
+
+  /** The [Infrastructure] used. */
+  val infrastructure: Infrastructure
+
+  /** The [LocalizedStrings] applied to the GUI. */
+  val strings: LocalizedStrings
+
+  /** The default [AuthenticationUser], which may be authenticated or not. */
+  val user: AuthenticationUser
+
+  // Destructuring-friendly syntax.
+  operator fun component1(): Facades = facades
+  operator fun component2(): Infrastructure = infrastructure
+  operator fun component3(): LocalizedStrings = strings
+  operator fun component4(): AuthenticationUser = user
+}
+
 /**
- * A class representing a standard testing environment.
+ * An implementation of [TestEnvironment].
  *
- * @property facades the [Facades] used.
- * @property infrastructure the [Infrastructure] used.
- * @property strings the [LocalizedStrings] applied to the GUI.
- * @property user the default [AuthenticatedUser].
+ * @see TestEnvironment
  */
-data class TestEnvironment(
-    val facades: Facades,
-    val infrastructure: Infrastructure,
-    val strings: LocalizedStrings,
-    val user: AuthenticatedUser,
-)
+private class ActualTestEnvironment(
+    override val facades: Facades,
+    override val infrastructure: Infrastructure,
+    override val strings: LocalizedStrings,
+    override val user: AuthenticationUser,
+) : TestEnvironment
+
+/** An interface representing a testing environment with an authenticated user. */
+interface AuthenticatedTestEnvironment : TestEnvironment {
+  override val user: AuthenticatedUser
+  override operator fun component4(): AuthenticatedUser = user
```

Comment:
[Question] Why is this function called component4?

Reply:
It's an operator which allows for the destructuring syntax.